lib/auth/windows/certificate_authority.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package windows
+
+import (
+	"bytes"
+	"context"
+	"encoding/pem"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+)
+
+// NewCertificateAuthority returns a new structure for modifying windows certificates in a windows CA
+func NewCertificateAuthority(clusterName string,
+	ldapConfig LDAPConfig,
+	ldapClient *LDAPClient,
+	ap auth.WindowsDesktopAccessPoint,
+	log logrus.FieldLogger) *CertificateAuthority {
```

Comment:
nit: Why not just: 
```suggestion
func NewCertificateAuthority(cfg CertificateAuthorityConfig) (*CertificateAuthority, error) {
 if err := cfg.ValidateAndSetDefault(); err != nil {
     return nil, err
 }
 return CertificateAuthority{cfg: cfg}, nil
}
```
where `clusterName` and `lc` can be part of `CertificateAuthorityConfig`

Reply:
Sounds very reasonable.